### PR TITLE
Create Login Screen UI: Brach changed and bugs fixed according to review

### PR DESCRIPTION
**Pull request by mfd2144**

### CODE REVIEW

File: Driver/Driver/Scenes/HomeScreen/HomeViewProtocolts.swift [hunk 1]
```diff
@@ -1,5 +1,5 @@
 //
-//  HomeViewContracts.swift
+//  HomeViewProtocolts.swift
```
Comment (TolgaTaner) [inline Driver/Driver/Scenes/HomeScreen/HomeViewProtocolts.swift:2]: should be HomeViewProtocol. Please, don't rush while coding.

File: Driver/Driver/Helper/Animator.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            }
+            controller.view.addSubview(self.transparentView)
+            controller.view.addSubview(self.indicator)
+            controller.view.bringSubviewToFront(self.transparentView)
+            controller.view.bringSubviewToFront(self.indicator)
+        }
+    }
+    func hideAnimation() {
+        DispatchQueue.main.async( execute: {
+            self.transparentView.removeFromSuperview()
```
Comment (TolgaTaner) [inline Driver/Driver/Helper/Animator.swift:44]: be careful about retain cycle. should be handle with weak self

File: Driver/Driver/Scenes/LoginScreen/ViewController/LoginViewController.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    // MARK: Properties
+    var viewModel: LoginViewModelProtocol!
+    
+    // MARK: LifeCycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        userNameTextField.delegate = self
```
Comment (TolgaTaner) [inline Driver/Driver/Scenes/LoginScreen/ViewController/LoginViewController.swift:25]: both setting self should be in viewDidLoad(). We shouldn't set them each this view controller appears.

**Review by TolgaTaner — COMMENTED** [review]
gj mate. Generally, we may not add documentation blocks to protocols. It will be duplicate with where we conform it. Don't forget to open your pull request to development branch, next time.

File: Driver/Driver/Extenison/UIColor+.swift [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+//  Driver
+//
+//  Created by Fatih Doğan on 22.02.2023.
+//
+
+import UIKit.UIColor
+
+extension UIColor {
+    static var appBlue: UIColor {
+        UIColor(named: "AppBlue")! }
```
Comment (Racaycah) [inline Driver/Driver/Extenison/UIColor+.swift:12]: I don't know yet if we have a rule for force unwrapping, just noting this is a usage

File: Driver/Driver/Scenes/LoginScreen/LoginViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import Foundation
+
+final class LoginViewModel: LoginViewModelProtocol {
+    weak var delegate: LoginViewDelegate?
+    var router: LoginRouterProtocol!
+    
+    func login(userName: String?, password: String?) {
+        if let userName, let password, !userName.isEmpty, !password.isEmpty {
+            delegate?.handleOutput(.setSpinner(isLoading: true))
+            DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
```
Comment (Racaycah) [inline Driver/Driver/Scenes/LoginScreen/LoginViewModel.swift:17]: Could use a weak capture here

File: Driver/Driver/Scenes/LoginScreen/ViewController/LoginViewController.swift [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            passwordTextField.becomeFirstResponder()
+        } else {
+            textField.endEditing(true)
+        }
+        return true
+    }
+}
+extension LoginViewController {
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        view.endEditing(true)
```
Comment (Racaycah) [inline Driver/Driver/Scenes/LoginScreen/ViewController/LoginViewController.swift:59]: We should call super as the first thing here, [as suggested for the method](https://developer.apple.com/documentation/uikit/uiresponder/1621142-touchesbegan)

File: Driver/Driver/Extenison/UIColor+.swift (hunk 4 above)
Comment (TolgaTaner) [reply Driver/Driver/Extenison/UIColor+.swift:12]: we need to force here. We should see any exception like this in the development process. @Racaycah

**Comment by mfd2144** [comment]
I changed the codes based on your comments. I detect that viewControllers do not deallocate correctly (You can see it also from baseview deinit method). I did a leak test and everything seems fine. But I believe some references prevent viewcontrollers(SplashScreenViewControllers and LoginViewControllers ) deallocated  but I couldn't find them.

File: Driver/Driver/Extenison/UIColor+.swift (hunk 4 above)
Comment (Racaycah) [reply Driver/Driver/Extenison/UIColor+.swift]: It should either be this or assertionFailure for that, I'm okay with force unwrapping here 😄

**Review by Racaycah — APPROVED** [review]
Thank you for the effort! We should be tracking the memory leak investigation somewhere (cc @TolgaTaner )

File: Driver/Driver/Extenison/UIColor+.swift (hunk 4 above)
Comment (TolgaTaner) [reply Driver/Driver/Extenison/UIColor+.swift]: @Racaycah yeah, You are right. I guess, It is little bit doing over engineering for this project :D @Racaycah

**Review by TolgaTaner — APPROVED** [review]

**Review by TolgaTaner — APPROVED** [review]

**Pull request merged**